spec/inputs/jdbc_spec.rb

```diff
@@ -1,172 +1,208 @@
+      expect { plugin.register }.to_not raise_error
+      plugin.teardown
+    end
 
-  it "should properly schedule" do
-    settings = {"statement" => "SELECT 1 as num_param FROM SYSIBM.SYSDUMMY1", "schedule" => "* * * * * UTC"}
-    plugin = LogStash::Inputs::Jdbc.new(mixin_settings.merge(settings))
-    plugin.register
-    q = Queue.new
-    Timecop.travel(Time.new(2000))
-    Timecop.scale(60)
-    runner = Thread.new do
-      plugin.run(q)
-    end
-    sleep 3
-    plugin.teardown
-    runner.kill
-    runner.join
-    insist { q.size } == 2
-    Timecop.return
+    it "should teardown without raising exception" do
+      plugin.register
+      expect { plugin.teardown }.to_not raise_error
+    end
   end
 
-  it "should appropriately page table" do
-    require "sequel"
-    require "sequel/adapters/jdbc"
-    Jdbc::Derby.load_driver
-    @database = Sequel.connect(mixin_settings['jdbc_connection_string'], :user=> nil, :password=> nil)
-    @database.create_table :test_table do
-      DateTime :created_at
-      Integer :num
+  context "when passing parameters" do
+    let(:settings) do
+      {
+        "statement" => "SELECT :num_param as num_param FROM SYSIBM.SYSDUMMY1",
+        "parameters" => {"num_param" => 10}
+      }
     end
-    test_table = @database[:test_table]
-    settings = {"statement" => "SELECT * from test_table",
-                "jdbc_paging_enabled" => true, "jdbc_page_size" => 20}
-    plugin = LogStash::Inputs::Jdbc.new(mixin_settings.merge(settings))
-    plugin.register
-    q = Queue.new
 
-    NUM_ROWS = 1000
+    before do
+      plugin.register
+    end
+
+    after do
+      plugin.teardown
+    end
 
-    NUM_ROWS.times do
-      test_table.insert(:num => 1, :created_at => Time.now.utc)
+    it "should retrieve params correctly from Event" do
+      plugin.run(queue)
+      expect(queue.pop['num_param']).to eq(settings['parameters']['num_param'])
     end
+  end
 
-    plugin.run(q)
-    plugin.teardown
+  context "when scheduling" do
+    let(:settings) { {"statement" => "SELECT 1 as num_param FROM SYSIBM.SYSDUMMY1", "schedule" => "* * * * * UTC"} }
 
-    insist { q.size } == NUM_ROWS
+    it "should properly schedule" do
+      plugin.register
+      Timecop.travel(Time.new(2000))
+      Timecop.scale(60)
+      runner = Thread.new do
+        plugin.run(queue)
+      end
+      sleep 3
+      plugin.teardown
+      runner.kill
+      runner.join
+      expect(queue.size).to eq(2)
+      Timecop.return
+    end
 
-    @database.drop_table(:test_table)
   end
 
-  it "should successfully iterate table with respect to field values" do
-    require "sequel"
-    require "sequel/adapters/jdbc"
-    Jdbc::Derby.load_driver
-    @database = Sequel.connect(mixin_settings['jdbc_connection_string'], :user=> nil, :password=> nil)
-    @database.create_table :test_table do
-      DateTime :created_at
-      Integer :num
+  context "with specific database" do
```

Comment:
not sure how to clean up this context. should I split this out into its own `describe` block so that I can declare new `settings` in the same manner as the other tests using `let`?